### PR TITLE
Story updates

### PR DESCRIPTION
**Pull request by mdiller**
Switched to the new `building_kill` format in the objectives log, while retaining support for the old format

### CODE REVIEW

File: src/components/Match/MatchStory.jsx [hunk 1]
```diff
@@ -126,6 +126,15 @@ const formatList = (items, noneValue = []) => {
   }
 };
 
+// is given a hero in the format of npc_dota_hero_{heroname}, and returns the player corresponding to that hero, else undefined
+const heroToPlayer = (match, heroname) => match.players.find((player) => {
```
Comment (howardchung) [inline src/components/Match/MatchStory.jsx]: doesn't the event have player_slot that you can use?

File: src/components/Match/MatchStory.jsx [hunk 2]
```diff
@@ -326,41 +335,70 @@ class LanesEvent extends StoryEvent {
   }
 }
 
+// returnes a formatted template when given a TowerEvent or BarracksEvent
+const formatBuilding = (event) => {
+  let template = strings.story_building_destroy;
+  if (event.player) {
+    template = event.player.isRadiant === event.team ? strings.story_building_deny_player : strings.story_building_destroy_player;
```
Comment (howardchung) [inline src/components/Match/MatchStory.jsx]: team is a number, isRadiant is a boolean?  Do you need a cast here?

File: src/components/Match/MatchStory.jsx (hunk 1 above)
Comment (mdiller) [inline src/components/Match/MatchStory.jsx]: I was figuring that the new `key` field would be more reliable than player_slot. Should I change it back?

File: src/components/Match/MatchStory.jsx (hunk 2 above)
Comment (mdiller) [inline src/components/Match/MatchStory.jsx]: for events, `team` is a boolean where radiant is true, and dire is false. It's different from how they are passed in through the objectives array.

File: src/components/Match/MatchStory.jsx (hunk 1 above)
Comment (howardchung) [inline src/components/Match/MatchStory.jsx]: I think the player_slot is populated on the server side using similar logic, so probably easier not to reimplement it

Comment (mdiller) [inline src/components/Match/MatchStory.jsx]: sounds good

**Pull request merged**